scripts/rconnect.js

```diff
@@ -1,53 +1,58 @@
 function recursiveScan(ns, parent, server, target, route) {
-    const children = ns.scan(server);
-
-    for (let child of children) {
-        if (parent == child) {
-            continue;
-        }
-        if (child == target) {
-            route.unshift(child);
-            route.unshift(server);
-            return true;
-        }
-
-        if (recursiveScan(ns, server, child, target, route)) {
-            route.unshift(server);
-            return true;
-        }
-    }
-
-    return false;
+	const children = ns.scan(server);
+
+	for (let child of children) {
+		if (parent == child) {
+			continue;
+		}
+		if (child == target) {
+			route.unshift(child);
+			route.unshift(server);
+			return true;
+		}
+
+		if (recursiveScan(ns, server, child, target, route)) {
+			route.unshift(server);
+			return true;
+		}
+	}
+
+	return false;
 }
 
 /** @param {NS} ns */
 export async function main(ns) {
-    const args = ns.flags([["help", false]]);
+	const args = ns.flags([["help", false]]);
 
-    let route = [];
-    let server = args._[0];
+	let route = [];
+	let server = args._[0];
 
-    if (!server || args.help) {
-        ns.tprint("This script helps you find a server on the network and shows you the path to get to it.");
-        ns.tprint(`Usage: run ${ns.getScriptName()} SERVER`);
-        ns.tprint("Example:");
-        ns.tprint(`> run ${ns.getScriptName()} n00dles`);
-        return;
-    }
+	if (!server || args.help) {
+		ns.tprint("This script helps you find a server on the network and shows you the path to get to it.");
+		ns.tprint(`Usage: run ${ns.getScriptName()} SERVER`);
+		ns.tprint("Example:");
+		ns.tprint(`> run ${ns.getScriptName()} n00dles`);
+		return;
+	}
 
-    recursiveScan(ns, '', 'home', server, route);
+	recursiveScan(ns, '', 'home', server, route);
 
-    for (const i in route) {
+	for (const i in route) {
 
-        await ns.sleep(500);
+		await ns.sleep(500);
 
-        const extra = i > 0 ? "└ " : "";
+		const extra = i > 0 ? "└ " : "";
 
-        ns.tprint(`${" ".repeat(i)}${extra}${route[i]}`);
-    }
+		ns.tprint(`${" ".repeat(i)}${extra}${route[i]}`);
+	}
+
+	ns.tprint("------------------------------------------------")
+	const terminalInput = document.getElementById("terminal-input");
```

Comment:
here:
get the terminal
set the value
get the event handler 
simulate enter keypress